File: 102_binary_tree_level_order_traversal/levelOrder.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func levelOrder(root *TreeNode) [][]int {
    res := make([][]int, 0)
    bfs(root, 0, &res)
    return res
}

func bfs(root *TreeNode, level int, res *[][]int) {
    if root == nil {
        return
    }

    if len(*res) <= level {
        *res = append(*res, []int{})
    }

    (*res)[level] = append((*res)[level], root.Val)
    bfs(root.Left, level + 1, res)
    bfs(root.Right, level + 1, res)
}

File: 102_binary_tree_level_order_traversal/levelOrder.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        bfs(root, 0, res);
        return res;
    }

    private void bfs(TreeNode root, int level, List<List<Integer>> res) {
        if (root == null) return;
        if (res.size() <= level) res.add(new ArrayList<>());

        res.get(level).add(root.val);
        bfs(root.left, level + 1, res);
        bfs(root.right, level + 1, res);
    }
}